ui/src/app/components/cases/list.component.ts

```diff
@@ -437,4 +438,48 @@ export class TestCasesListComponent extends BaseComponent implements OnInit {
       this.userPreferenceService.save(userPreference).subscribe(res => this.userPreference = res)
     }
   }
+  public fetchLinkedTestCases(stepGroupId) {
+    let testCases: InfiniteScrollableDataSource;
+    testCases = new InfiniteScrollableDataSource(this.testCaseService, "workspaceVersionId:" + this.versionId + ",deleted:false,stepGroupId:" + stepGroupId);
+    waitTillRequestResponds();
+    let _this = this;
+
+    function waitTillRequestResponds() {
+      if (testCases.isFetching)
+        setTimeout(() => waitTillRequestResponds(), 100);
```

Comment:
is promise based approach doesn't workout to avoid explicit timeout operation?

Reply:
im not sure. the timeout was used in cloud and i asked vigneshwaran bro about this and he said its needed if the testcases is not fetched completely